sycl/test/basic_tests/vectors/vector_operators.cpp

```diff
@@ -0,0 +1,249 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+//==---------- vector_operators.cpp - SYCL vec<> operators test ------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#define SYCL_SIMPLE_SWIZZLES
+#include <CL/sycl.hpp>
+namespace s = cl::sycl;
+
+template <typename ResultVecT>
+void check_result_length_4(ResultVecT &res, ResultVecT &expected_res) {
+  assert(res.get_count() == 4 && expected_res.get_count() == 4);
+
+  assert(res.template swizzle<s::elem::s0>() ==
+         expected_res.template swizzle<s::elem::s0>());
+  assert(res.template swizzle<s::elem::s1>() ==
+         expected_res.template swizzle<s::elem::s1>());
+  assert(res.template swizzle<s::elem::s2>() ==
+         expected_res.template swizzle<s::elem::s2>());
+  assert(res.template swizzle<s::elem::s3>() ==
+         expected_res.template swizzle<s::elem::s3>());
+}
+
+int main() {
+
+  /* Separate checks for NumElements=1 edge case */
+
+  {
+    using vec_type = s::vec<s::cl_char, 1>;
+    vec_type res;
+    {
+      s::buffer<vec_type, 1> Buf(&res, s::range<1>(1));
+      s::queue Queue;
+      Queue.submit([&](s::handler &cgh) {
+        auto Acc = Buf.get_access<s::access::mode::write>(cgh);
+        cgh.single_task<class isequal_vec_op_1_elem>([=]() {
+          vec_type vec1(2);
+          vec_type vec2(2);
+          Acc[0] = vec1 == vec2;
+        });
+      });
+    }
+    // 1-element vector operators currently follow
+    // vector 0/-1 logic
+    vec_type expected_res(-1);
+    assert(res.template swizzle<cl::sycl::elem::s0>() ==
+           expected_res.template swizzle<cl::sycl::elem::s0>());
+  }
+
+  {
+    using vec_type = s::vec<s::cl_char, 1>;
+    vec_type res;
+    {
+      s::buffer<vec_type, 1> Buf(&res, s::range<1>(1));
+      s::queue Queue;
+      Queue.submit([&](s::handler &cgh) {
+        auto Acc = Buf.get_access<s::access::mode::write>(cgh);
+        cgh.single_task<class isequal_vec_op_1_elem_scalar>([=]() {
+          vec_type vec(2);
+          s::cl_char rhs_scalar = 2;
+          Acc[0] = vec == rhs_scalar;
+        });
+      });
+    }
+    // 1-element vector operators currently follow
```

Comment:
What do you mean by `currently`? Will this behavior be changed and why? What about 4-element vectors?